### PR TITLE
Apply updates from yorkie-js-sdk 0.7.11

### PR DESCRIPTION
**Pull request by longhoang-lewy**
**What this PR does / why we need it**:

Syncs `yorkie-ios-sdk` with `yorkie-js-sdk` **v0.7.11**. Three of the four upstream changes are iOS-relevant and ported here (one commit each, subject = the upstream PR title); the React-only change is skipped.

- **Introduce treelist to optimize random access in Array** (js#1269) — new `Sources/Util/TreeList.swift`, a Swift port of the upstream Left-leaning Red-Black order-statistic tree (per-node `weight` = live count, `count` = total incl. tombstones). `RGATreeList` is rewired off `SplayTree` onto `TreeList`. As part of this change, `RemoveOperation` now reads the element's sub-path/index **before** deleting it — required because `TreeList.indexOf` returns `-1` for tombstones (matching JS), which also fixes a latent iOS index reporting bug the old `SplayTree` had masked.
- **Stop disableGC clients from accumulating per-Change VV** (js#1270) — adds `ChangeID.syncLamport`; opt-out documents advance only the lamport clock instead of merging every remote actor's version vector, keeping each local Change's VV at O(1). `opts.disableGC` (construction, gates GC) and the attach-time `disableGC` (gates the sync path) are kept as two independent fields, mirroring JS.
- **Skip bare position nodes in RGATreeList.getLast** (js#1272).

**Which issue(s) this PR fixes**:

N/A

**Special notes for your reviewer**:

- Reviewed by the `critic-reviewer` agent against `refs/tags/v0.7.11`; its one High finding (initially conflating the two `disableGC` fields) was fixed and is guarded by a new integration test.
- Lint clean with the CI-pinned toolchain: **SwiftFormat 0.55.6** and **SwiftLint 0.58.2 `--strict`** (0 violations).
- `swift build` + full unit suite green (**594 tests**); the array / GC / Document / disableGC **integration** suites also run green against a local server (incl. the js#1270 "opt-out VV stays size 1" parity test and a new "disableGC attach does not disable local GC" regression test).
- The CI `yorkie` server pin is bumped to **0.7.11** (`swift.yml`, `swift-integration.yml`, both `docker-compose` files); `Sources/Version.swift` → `0.7.11`.

**Does this PR introduce a user-facing change?**:

```release-note
Clients attaching with `disableGC: true` no longer accumulate per-Change version-vector entries under multi-actor workloads (O(1) instead of O(num_actors)).
```

**Additional documentation**:

```docs

```

**Checklist**:
- [x] Added relevant tests or not required
- [x] Didn't break anything

## Summary by CodeRabbit

* **New Features**
  * Added improved synchronization behavior for documents when garbage collection is disabled.
  * Introduced an optimized order-statistics list structure to improve indexing and move/remove behavior.

* **Bug Fixes**
  * Fixed list tail reporting after move operations by skipping dead/bare trailing positions.
  * Corrected element removal bookkeeping to ensure indices/path info stays consistent.

* **Tests**
  * Added unit and integration coverage for disableGC synchronization and list/index edge cases.

* **Chores**
  * Updated CI and Docker images to the latest 0.7.11 release and refreshed the changelog.

### CODE REVIEW

File: Sources/Document/Document.swift [hunk 1]
```diff
@@ -662,13 +669,22 @@ public class Document: Attachable {
...
     }
 
+    /**
+     * `setDisableGC` records whether this document participates in GC. The
+     * client calls this on attach so subsequent `applyChanges` runs use the
+     * lamport-only sync path and GC is skipped.
+     */
+    func setDisableGC(_ disableGC: Bool) {
+        self.disableGC = disableGC
+    }
```
[the remark this answers is not in the document]
Comment (longhoang-lewy) [reply Sources/Document/Document.swift:680]: Fixed in ff24e0cff9 — the docstring now states `setDisableGC` only selects the lamport-only sync path in `applyChanges` and is distinct from `optionDisableGC`, which independently gates `garbageCollect` (so this flag does not disable GC).

File: Sources/Document/Document.swift [hunk 2]
```diff
@@ -877,7 +893,12 @@ public class Document: Attachable {
                 }
             }
 
-            self.changeID = self.changeID.syncClocks(with: change.id)
+            // Opt-out (disableGC) documents advance only the lamport clock and do
+            // not merge remote actors' version vectors, keeping each subsequent
+            // local Change's VV at O(1). See ``setDisableGC(_:)``.
+            self.changeID = self.disableGC
+                ? self.changeID.syncLamport(with: change.id)
+                : self.changeID.syncClocks(with: change.id)
```
[the remark this answers is not in the document]
Comment (longhoang-lewy) [reply Sources/Document/Document.swift:902]: Declining — this would diverge from the reference SDK. In yorkie-js-sdk v0.7.11, `applySnapshot` uses `changeID.setClocks(snapshotVector.maxLamport(), snapshotVector)` with **no** `disableGC` branch (document.ts:1384); only `applyChange` has the `syncLamport` path. The O(1) per-Change VV guarantee for opt-out clients on snapshots is provided **server-side** — the server returns a single-entry VV in the snapshot response for disableGC clients, so `setClocks` merges a size-1 vector. Putting the client's `applySnapshot` on the lamport-only path would break wire/behavioral parity with the JS and Android SDKs. iOS already mirrors JS here (`setClocks`).

File: Sources/Util/TreeList.swift [hunk 3]
```diff
@@ -0,0 +1,447 @@
...
+/// It tracks two aggregates over its subtree:
+///   - `weight`: number of non-removed (live) nodes (logical index)
+///   - `count`: total nodes including tombstones (structural index)
+final class TreeListNode<V: TreeListValue> {
+    /// The value carried by this node.
+    ///
+    /// The reference is strong: for the RGATreeList integration the value
+    /// (`RGATreeListNode`) owns its index node, and this back-reference is
+    /// released via `TreeList.delete` when the position is physically purged.
+    let value: V
```
[the remark this answers is not in the document]
Comment (longhoang-lewy) [reply Sources/Util/TreeList.swift:39]: Fixed in ff24e0cff9 — added a `deinit` to `RGATreeList` that walks the linked list once and clears `previous`/`next`/`indexNode` on every node, breaking both the doubly-linked-list cycle and the `indexNode`↔`value` cycle so the node graph deallocates with the list. (Note: the `previous`/`next` cycle pre-dated this PR under the old `SplayTree`; the teardown now resolves that too.)

**Review by ngocann — APPROVED** [review]
